Mlem/Views/Shared/Instance/InstanceUptimeView.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  InstanceUptimeView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 28/01/2024.
+//
+
+import SwiftUI
+import Charts
+import Dependencies
+import Foundation
+
+struct InstanceUptimeView: View {
+    @Dependency(\.errorHandler) var errorHandler
+    
+    @Environment(\.accessibilityDifferentiateWithoutColor) var diffWithoutColor: Bool
+    @Environment(\.colorScheme) var colorScheme
+    
+    @State var showingExactTime: Bool = false
+    
+    let instance: InstanceModel
+    let uptimeData: UptimeData
+    
+    @ViewBuilder
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            section { summary }
+            .padding(.top, 16)
+            .padding(.bottom, 10)
+            section("Recent Checks") {
+                recentChecks
+                    .padding(.horizontal)
+                    .padding(.vertical, 15)
+            }
+            .padding(.top, 20)
+            footnote("Mlem will refresh this automatically every 30 seconds.")
+            .padding(.top, 8)
+            .padding(.leading, 6)
+            .padding(.bottom, 30)
+            section("Response Time") {
+                VStack(alignment: .leading, spacing: 4) {
+                    responseTimeChart
+                        .padding(.horizontal, 20)
+                    footnote("Mean average: \(uptimeData.results.map(\.durationMs).reduce(0, +) / uptimeData.results.count)ms")
+                        .padding(.leading, 20)
+                }
+                .padding(.top, 17)
+                .padding(.bottom, 8)
+            }
+            section("Incidents", spacing: 0) {
+                ForEach(uptimeData.downtimes) { event in
```

Comment:
Maybe only show outages from the past day, and hide the rest in a disclosure group? As-is it's a super long stack of low-impact data. It might also be valuable to add a filter to show only medium to high severity incidents?

Reply:
That could work. 

I'm not exactly sure how much I trust the incidents list to be honest - [lemmy-status has been reporting programming.dev as unhealthy for 3 days](https://lemmy-status.org/endpoints/_programming-dev), but it's working just fine. Beehaw has a similar issue 🤔 

What do we do in that situation?

Reply:
I think augmenting the footer to read "Uptime data fetched from lemmy-status.org, and may contain inaccuracies" would be sufficient. If we wanted to be a little fancier we could add some logic to detect discrepancies between the most recent incident and the current status, and if one is detected add a footer to that incident reading "Discrepancy detected" with a tappable element to open a popup explaining that the incident and the current active status conflict, and that the user should trust the active status.

Reply:
Hmm. Maybe it would be better to just hide the faulty incident if it's in conflict with the actual status?

Reply:
That works for me

Reply:
How's this?